lang/resources/org/partiql/type-domains/partiql.ion

```diff
@@ -210,12 +210,68 @@ may then be further optimized by selecting better implementations of each operat
             // UNPIVOT <expr> [AS <id>] [AT <id>] [BY <id>]
             (unpivot expr::expr as_alias::(? symbol) at_alias::(? symbol) by_alias::(? symbol))
 
-            // <from_source> JOIN [INNER | LEFT | RIGHT | FULL] <from_source> ON <expr>
-            (join type::join_type left::from_source right::from_source predicate::(? expr)))
+             // <from_source> JOIN [INNER | LEFT | RIGHT | FULL] <from_source> ON <expr>
+             (join type::join_type left::from_source right::from_source predicate::(? expr))
+
+             // <expr> MATCH <graph_pattern>
+             (graph_match expr::expr graph_expr::graph_match_expr))
 
         // Indicates the logical type of join.
         (sum join_type (inner) (left) (right) (full))
 
+        // The direction of an edge
+        // | Orientation               | Edge pattern | Abbreviation |
+        // |---------------------------+--------------+--------------|
+        // | Pointing left             | <−[ spec ]−  | <−           |
+        // | Undirected                | ~[ spec ]~   | ~            |
+        // | Pointing right            | −[ spec ]−>  | −>           |
+        // | Left or undirected        | <~[ spec ]~  | <~           |
+        // | Undirected or right       | ~[ spec ]~>  | ~>           |
+        // | Left or right             | <−[ spec ]−> | <−>          |
+        // | Left, undirected or right | −[ spec ]−   | −            |
+        //
+        // Fig. 5. Table of edge patterns:
+        // https://arxiv.org/abs/2112.06217
+        (sum graph_match_direction
```

Comment:
nit: could call this `graph_edge_direction` since all the variants relate to edge direction

Reply:
I'm using `graph_match` as a prefix throughout to signify the parts of the grammar that deal with graph patterns/matching. So, read this more as though it were `graph_match::direction`